src/connector.rs

```diff
@@ -147,3 +133,76 @@ impl<T> fmt::Debug for HttpsConnector<T> {
             .finish()
     }
 }
+
+/// The default server name resolver, which uses the hostname in the URI.
+#[derive(Default)]
+pub struct DefaultServerNameResolver(());
+
+impl DefaultServerNameResolver {
```

Comment:
Nit: I would prefer ditching this `impl` and relying on `Default` only.

Reply:
removed!